src/hotspot/share/gc/shared/memAllocator.hpp

```diff
@@ -98,6 +98,7 @@ class ObjArrayAllocator: public MemAllocator {
   const int  _length;
   const bool _do_zero;
 
+  void mem_zap_alignment_padding(HeapWord* mem) const PRODUCT_RETURN;
```

Comment:
Both methods are about alignment-padding, one for array-element type, the other for obj itself. Maybe call it "mem_zap_start_padding" to make them more symmetric?

Reply:
I am not opposed to that name. Only confusion could be that someone thinks that it is before the header, but the current name is not any better. I'll take your suggestion.